File: official/docs/node/pickups/create_pickup.js
const Easypost = require('@easypost/api');
const api = new Easypost('EASYPOST_API_KEY');

const address = 'adr_...';
const shipment = 'shp_...';

const pickup = new api.Pickup({
  address,
  shipment,
  reference: 'my-first-pickup',
  min_datetime: '2014-10-21 10:30:00',
  max_datetime: '2014-10-22 10:30:00',
  is_account_address: false,
  instructions: 'Special pickup instructions',
});

pickup.save().then(console.log);

File: official/docs/csharp/pickups/create-pickup.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Xunit;
using Xunit.Sdk;
using Newtonsoft.Json;
using EasyPost;

namespace EasyPostExamples;

public class Examples
{
    [Fact]
    public async Task CreatePickup()
    {
        string apiKey = Environment.GetEnvironmentVariable("EASYPOST_API_KEY")!;

        EasyPost.ClientManager.SetCurrent(apiKey);

        Pickup pickup = await Pickup.Create(new Dictionary<string, object>()
        {
            {
                "reference", "my-first-pickup"
            },
            {
                "min_datetime", "2022-10-01 10:30:00"
            },
            {
                "max_datetime", "2022-10-01 17:30:00"
            },
            {
                "shipment", "shp_..."
            },
            {
                "address", "adr_..."
            },
            {
                "is_account_address", false
            },
            {
                "instructions", "Special pickup instructions"
            },

        });

        new TestOutputHelper().WriteLine(JsonConvert.SerializeObject(pickup, Formatting.Indented));
    }
}
